Subject: gtod: optimize
From: Ingo Molnar <mingo@elte.hu>
Date: Wed Feb 04 00:02:44 CET 2009

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 kernel/timer.c |    7 +++++++
 1 file changed, 7 insertions(+)

Index: linux-2.6-tip/kernel/timer.c
===================================================================
--- linux-2.6-tip.orig/kernel/timer.c
+++ linux-2.6-tip/kernel/timer.c
@@ -1277,6 +1277,13 @@ static inline void update_times(void)
 	static unsigned long last_tick = INITIAL_JIFFIES;
 	unsigned long ticks, flags;
 
+	/*
+	 * Dont take the xtime_lock from every CPU in
+	 * every tick - only when needed:
+	 */
+	if (jiffies == last_tick)
+		return;
+
 	write_seqlock_irqsave(&xtime_lock, flags);
 	ticks = jiffies - last_tick;
 	if (ticks) {
