Subject: hrtimer-fix-wait-for-hrtimer.patch
From: Thomas Gleixner <tglx@linutronix.de>
Date: Wed, 04 Feb 2009 22:50:43 +0100

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 kernel/hrtimer.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

Index: linux-2.6-tip/kernel/hrtimer.c
===================================================================
--- linux-2.6-tip.orig/kernel/hrtimer.c
+++ linux-2.6-tip/kernel/hrtimer.c
@@ -854,7 +854,7 @@ void hrtimer_wait_for_timer(const struct
 {
 	struct hrtimer_clock_base *base = timer->base;
 
-	if (base && base->cpu_base)
+	if (base && base->cpu_base && !hrtimer_hres_active(base->cpu_base))
 		wait_event(base->cpu_base->wait,
 				!(timer->state & HRTIMER_STATE_CALLBACK));
 }
@@ -891,8 +891,6 @@ static void __remove_hrtimer(struct hrti
 		rb_erase(&timer->node, &base->active);
 	}
 	timer->state = newstate;
-
-	wake_up_timer_waiters(base->cpu_base);
 }
 
 /*
@@ -1420,6 +1418,8 @@ void hrtimer_run_queues(void)
 		}
 		spin_unlock(&cpu_base->lock);
 	}
+
+	wake_up_timer_waiters(cpu_base);
 }
 
 /*
