Subject: i386: mark atomic irq ops raw
From: Ingo Molnar <mingo@elte.hu>
Date: Wed Feb 04 00:03:14 CET 2009

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 arch/x86/include/asm/atomic_32.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6-tip/arch/x86/include/asm/atomic_32.h
===================================================================
--- linux-2.6-tip.orig/arch/x86/include/asm/atomic_32.h
+++ linux-2.6-tip/arch/x86/include/asm/atomic_32.h
@@ -180,10 +180,10 @@ static inline int atomic_add_return(int 
 
 #ifdef CONFIG_M386
 no_xadd: /* Legacy 386 processor */
-	local_irq_save(flags);
+	raw_local_irq_save(flags);
 	__i = atomic_read(v);
 	atomic_set(v, i + __i);
-	local_irq_restore(flags);
+	raw_local_irq_restore(flags);
 	return i + __i;
 #endif
 }
