Subject: lockdep: atomic fixup
From: Ingo Molnar <mingo@elte.hu>
Date: Wed Feb 04 00:01:59 CET 2009

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 lib/locking-selftest.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6-tip/lib/locking-selftest.c
===================================================================
--- linux-2.6-tip.orig/lib/locking-selftest.c
+++ linux-2.6-tip/lib/locking-selftest.c
@@ -948,7 +948,7 @@ static void dotest(void (*testcase_fn)(v
 	unsigned long saved_preempt_count = preempt_count();
 	int expected_failure = 0;
 #if defined(CONFIG_DEBUG_PREEMPT) && defined(CONFIG_DEBUG_RT_MUTEXES)
-        int saved_lock_count = current->lock_count;
+        atomic_t saved_lock_count = current->lock_count;
 #endif
 
 	WARN_ON(irqs_disabled());
