Subject: mm-lock-break-on-bulk.patch
From: Thomas Gleixner <tglx@linutronix.de>
Date: Mon, 16 Mar 2009 17:06:15 +0100

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 mm/page_alloc.c |    1 +
 1 file changed, 1 insertion(+)

Index: linux-2.6-tip/mm/page_alloc.c
===================================================================
--- linux-2.6-tip.orig/mm/page_alloc.c
+++ linux-2.6-tip/mm/page_alloc.c
@@ -578,6 +578,7 @@ static void free_pages_bulk(struct zone 
 		/* have to delete it as __free_one_page list manipulates */
 		list_del(&page->lru);
 		__free_one_page(page, zone, order);
+		cond_resched_lock(&zone->lock);
 	}
 	spin_unlock(&zone->lock);
 }
