Subject: msi: suspend resume workaround
From: Ingo Molnar <mingo@elte.hu>
Date: Wed Feb 04 00:03:13 CET 2009

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 drivers/pci/msi.c |    4 ++++
 1 file changed, 4 insertions(+)

Index: linux-2.6-tip/drivers/pci/msi.c
===================================================================
--- linux-2.6-tip.orig/drivers/pci/msi.c
+++ linux-2.6-tip/drivers/pci/msi.c
@@ -323,6 +323,10 @@ static void __pci_restore_msi_state(stru
 		return;
 
 	entry = get_irq_msi(dev->irq);
+	if (!entry) {
+		WARN_ON(1);
+		return;
+	}
 	pos = entry->msi_attrib.pos;
 
 	pci_intx_for_msi(dev, 0);
