Subject: powerpc-raw-spin-unlock-wait-disable-preemption.patch
From: Thomas Gleixner <tglx@linutronix.de>
Date: Mon, 23 Mar 2009 22:55:53 +0100

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 arch/powerpc/lib/locks.c |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-2.6-tip/arch/powerpc/lib/locks.c
===================================================================
--- linux-2.6-tip.orig/arch/powerpc/lib/locks.c
+++ linux-2.6-tip/arch/powerpc/lib/locks.c
@@ -86,8 +86,10 @@ void __raw_spin_unlock_wait(__raw_spinlo
 {
 	while (lock->slock) {
 		HMT_low();
+		preempt_disable();
 		if (SHARED_PROCESSOR)
 			__spin_yield(lock);
+		preempt_enable();
 	}
 	HMT_medium();
 }
