Subject: powerpc-compat-semaphore.patch
From: Thomas Gleixner <tglx@linutronix.de>
Date: Fri, 20 Mar 2009 17:34:03 +0100

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 drivers/macintosh/adb.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6-tip/drivers/macintosh/adb.c
===================================================================
--- linux-2.6-tip.orig/drivers/macintosh/adb.c
+++ linux-2.6-tip/drivers/macintosh/adb.c
@@ -83,7 +83,7 @@ static struct adb_driver *adb_controller
 BLOCKING_NOTIFIER_HEAD(adb_client_list);
 static int adb_got_sleep;
 static int adb_inited;
-static DECLARE_MUTEX(adb_probe_mutex);
+static COMPAT_DECLARE_MUTEX(adb_probe_mutex);
 static int sleepy_trackpad;
 static int autopoll_devs;
 int __adb_probe_sync;
