Subject: preempt: realtime sched cpupri
From: Ingo Molnar <mingo@elte.hu>
Date: Wed Feb 04 00:02:54 CET 2009

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 kernel/sched_cpupri.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6-tip/kernel/sched_cpupri.h
===================================================================
--- linux-2.6-tip.orig/kernel/sched_cpupri.h
+++ linux-2.6-tip/kernel/sched_cpupri.h
@@ -12,7 +12,7 @@
 /* values 2-101 are RT priorities 0-99 */
 
 struct cpupri_vec {
-	spinlock_t lock;
+	raw_spinlock_t lock;
 	int        count;
 	cpumask_var_t mask;
 };
