Subject: preempt: realtime supress rtc printk
From: Ingo Molnar <mingo@elte.hu>
Date: Wed Feb 04 00:02:45 CET 2009

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 drivers/char/rtc.c |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-2.6-tip/drivers/char/rtc.c
===================================================================
--- linux-2.6-tip.orig/drivers/char/rtc.c
+++ linux-2.6-tip/drivers/char/rtc.c
@@ -1201,10 +1201,12 @@ static void rtc_dropped_irq(unsigned lon
 
 	spin_unlock_irq(&rtc_lock);
 
+#ifndef CONFIG_PREEMPT_RT
 	if (printk_ratelimit()) {
 		printk(KERN_WARNING "rtc: lost some interrupts at %ldHz.\n",
 			freq);
 	}
+#endif
 
 	/* Now we have new data */
 	wake_up_interruptible(&rtc_wait);
