Subject: patches/random-driver-latency-fix.patch


Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 drivers/char/random.c |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

Index: linux-2.6-tip/drivers/char/random.c
===================================================================
--- linux-2.6-tip.orig/drivers/char/random.c
+++ linux-2.6-tip/drivers/char/random.c
@@ -623,8 +623,11 @@ static void add_timer_randomness(struct 
 	preempt_disable();
 	/* if over the trickle threshold, use only 1 in 4096 samples */
 	if (input_pool.entropy_count > trickle_thresh &&
-	    (__get_cpu_var(trickle_count)++ & 0xfff))
-		goto out;
+	    (__get_cpu_var(trickle_count)++ & 0xfff)) {
+		preempt_enable();
+		return;
+	}
+	preempt_enable();
 
 	sample.jiffies = jiffies;
 	sample.cycles = get_cycles();
@@ -666,8 +669,6 @@ static void add_timer_randomness(struct 
 		credit_entropy_bits(&input_pool,
 				    min_t(int, fls(delta>>1), 11));
 	}
-out:
-	preempt_enable();
 }
 
 void add_input_randomness(unsigned int type, unsigned int code,
