Subject: revert-bkl-cleanup.patch
From: Thomas Gleixner <tglx@linutronix.de>
Date: Mon, 09 Mar 2009 12:20:55 +0100

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 lib/kernel_lock.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6-tip/lib/kernel_lock.c
===================================================================
--- linux-2.6-tip.orig/lib/kernel_lock.c
+++ linux-2.6-tip/lib/kernel_lock.c
@@ -39,7 +39,7 @@ static  __cacheline_aligned_in_smp DEFIN
 int __lockfunc __reacquire_kernel_lock(void)
 {
 	while (!_raw_spin_trylock(&kernel_flag)) {
-		if (need_resched())
+		if (test_thread_flag(TIF_NEED_RESCHED))
 			return -EAGAIN;
 		cpu_relax();
 	}
