Subject: rt: mingo kernel_sem fix
From: Ingo Molnar <mingo@elte.hu>
Date: Sun Feb 08 19:33:10 CET 2009

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 kernel/rtmutex.c |    4 ++++
 1 file changed, 4 insertions(+)

Index: linux-2.6-tip/kernel/rtmutex.c
===================================================================
--- linux-2.6-tip.orig/kernel/rtmutex.c
+++ linux-2.6-tip/kernel/rtmutex.c
@@ -889,6 +889,7 @@ static inline int rt_release_bkl(struct 
 {
 	int saved_lock_depth = current->lock_depth;
 
+#ifdef CONFIG_LOCK_KERNEL
 	current->lock_depth = -1;
 	/*
 	 * try_to_take_lock set the waiters, make sure it's
@@ -900,14 +901,17 @@ static inline int rt_release_bkl(struct 
 	up(&kernel_sem);
 
 	spin_lock_irq(&lock->wait_lock);
+#endif
 
 	return saved_lock_depth;
 }
 
 static inline void rt_reacquire_bkl(int saved_lock_depth)
 {
+#ifdef CONFIG_LOCK_KERNEL
 	down(&kernel_sem);
 	current->lock_depth = saved_lock_depth;
+#endif
 }
 
 /*
