Subject: RT_utsname
From: Ingo Molnar <mingo@elte.hu>
Date: Wed Feb 04 00:02:29 CET 2009

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 init/Makefile       |    3 ++-
 scripts/mkcompile_h |    4 +++-
 2 files changed, 5 insertions(+), 2 deletions(-)

Index: linux-2.6-tip/init/Makefile
===================================================================
--- linux-2.6-tip.orig/init/Makefile
+++ linux-2.6-tip/init/Makefile
@@ -33,4 +33,5 @@ silent_chk_compile.h = :
 include/linux/compile.h: FORCE
 	@$($(quiet)chk_compile.h)
 	$(Q)$(CONFIG_SHELL) $(srctree)/scripts/mkcompile_h $@ \
-	"$(UTS_MACHINE)" "$(CONFIG_SMP)" "$(CONFIG_PREEMPT)" "$(CC) $(KBUILD_CFLAGS)"
+	"$(UTS_MACHINE)" "$(CONFIG_SMP)" "$(CONFIG_PREEMPT)" "$(CONFIG_PREEMPT_RT)" \
+	"$(CC) $(KBUILD_CFLAGS)"
Index: linux-2.6-tip/scripts/mkcompile_h
===================================================================
--- linux-2.6-tip.orig/scripts/mkcompile_h
+++ linux-2.6-tip/scripts/mkcompile_h
@@ -2,7 +2,8 @@ TARGET=$1
 ARCH=$2
 SMP=$3
 PREEMPT=$4
-CC=$5
+PREEMPT_RT=$5
+CC=$6
 
 vecho() { [ "${quiet}" = "silent_" ] || echo "$@" ; }
 
@@ -45,6 +46,7 @@ UTS_VERSION="#$VERSION"
 CONFIG_FLAGS=""
 if [ -n "$SMP" ] ; then CONFIG_FLAGS="SMP"; fi
 if [ -n "$PREEMPT" ] ; then CONFIG_FLAGS="$CONFIG_FLAGS PREEMPT"; fi
+if [ -n "$PREEMPT_RT" ] ; then CONFIG_FLAGS="$CONFIG_FLAGS RT"; fi
 UTS_VERSION="$UTS_VERSION $CONFIG_FLAGS $TIMESTAMP"
 
 # Truncate to maximum length
