Subject: rtmutex-mutex-fixups.patch
From: Thomas Gleixner <tglx@linutronix.de>
Date: Tue, 24 Mar 2009 21:50:26 +0100

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 include/linux/mutex.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6-tip/include/linux/mutex.h
===================================================================
--- linux-2.6-tip.orig/include/linux/mutex.h
+++ linux-2.6-tip/include/linux/mutex.h
@@ -214,6 +214,8 @@ extern int __must_check mutex_lock_killa
 extern int mutex_trylock(struct mutex *lock);
 extern void mutex_unlock(struct mutex *lock);
 
+#endif /* !PREEMPT_RT */
+
 /**
  * atomic_dec_and_mutex_lock - return holding mutex if we dec to 0
  * @cnt: the atomic which we are to dec
@@ -237,6 +239,4 @@ static inline int atomic_dec_and_mutex_l
 	return 1;
 }
 
-#endif /* !PREEMPT_RT */
-
 #endif
