Subject: smp: processor id fixups
From: Ingo Molnar <mingo@elte.hu>
Date: Wed Feb 04 00:02:32 CET 2009

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 include/linux/netpoll.h |    2 +-
 kernel/workqueue.c      |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6-tip/include/linux/netpoll.h
===================================================================
--- linux-2.6-tip.orig/include/linux/netpoll.h
+++ linux-2.6-tip/include/linux/netpoll.h
@@ -77,7 +77,7 @@ static inline void *netpoll_poll_lock(st
 	rcu_read_lock(); /* deal with race on ->npinfo */
 	if (dev && dev->npinfo) {
 		spin_lock(&napi->poll_lock);
-		napi->poll_owner = smp_processor_id();
+		napi->poll_owner = raw_smp_processor_id();
 		return napi;
 	}
 	return NULL;
Index: linux-2.6-tip/kernel/workqueue.c
===================================================================
--- linux-2.6-tip.orig/kernel/workqueue.c
+++ linux-2.6-tip/kernel/workqueue.c
@@ -209,7 +209,7 @@ static void delayed_work_timer_fn(unsign
 	struct cpu_workqueue_struct *cwq = get_wq_data(&dwork->work);
 	struct workqueue_struct *wq = cwq->wq;
 
-	__queue_work(wq_per_cpu(wq, smp_processor_id()), &dwork->work);
+	__queue_work(wq_per_cpu(wq, raw_smp_processor_id()), &dwork->work);
 }
 
 /**
