Subject: spinlock: trylock cleanup sungem
From: Ingo Molnar <mingo@elte.hu>
Date: Wed Feb 04 00:03:15 CET 2009

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 drivers/net/sungem.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

Index: linux-2.6-tip/drivers/net/sungem.c
===================================================================
--- linux-2.6-tip.orig/drivers/net/sungem.c
+++ linux-2.6-tip/drivers/net/sungem.c
@@ -1032,10 +1032,8 @@ static int gem_start_xmit(struct sk_buff
 			(csum_stuff_off << 21));
 	}
 
-	local_irq_save(flags);
-	if (!spin_trylock(&gp->tx_lock)) {
+	if (!spin_trylock_irqsave(&gp->tx_lock, flags)) {
 		/* Tell upper layer to requeue */
-		local_irq_restore(flags);
 		return NETDEV_TX_LOCKED;
 	}
 	/* We raced with gem_do_stop() */
