Subject: user: no irq disable
From: Ingo Molnar <mingo@elte.hu>
Date: Wed Feb 04 00:02:19 CET 2009

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 kernel/user.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6-tip/kernel/user.c
===================================================================
--- linux-2.6-tip.orig/kernel/user.c
+++ linux-2.6-tip/kernel/user.c
@@ -297,14 +297,14 @@ static void cleanup_user_struct(struct w
 	 */
 	uids_mutex_lock();
 
-	local_irq_save(flags);
+	local_irq_save_nort(flags);
 
 	if (atomic_dec_and_lock(&up->__count, &uidhash_lock)) {
 		uid_hash_remove(up);
 		remove_user = 1;
 		spin_unlock_irqrestore(&uidhash_lock, flags);
 	} else {
-		local_irq_restore(flags);
+		local_irq_restore_nort(flags);
 	}
 
 	if (!remove_user)
