Subject: add -rt version tag
From: Ingo Molnar <mingo@elte.hu>
Date: Mon Feb 09 23:09:57 CET 2009

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6-tip/Makefile
===================================================================
--- linux-2.6-tip.orig/Makefile
+++ linux-2.6-tip/Makefile
@@ -1,7 +1,7 @@
 VERSION = 2
 PATCHLEVEL = 6
 SUBLEVEL = 29
-EXTRAVERSION =
+EXTRAVERSION = -rt1
 NAME = Temporary Tasmanian Devil
 
 # *DOCUMENTATION*
