Subject: rtc: rtc-tegra: Use proper lock initializer
From: Thomas Gleixner <tglx@linutronix.de>
Date: Fri, 22 Jul 2011 08:27:11 +0200

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 drivers/rtc/rtc-tegra.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6/drivers/rtc/rtc-tegra.c
===================================================================
--- linux-2.6.orig/drivers/rtc/rtc-tegra.c
+++ linux-2.6/drivers/rtc/rtc-tegra.c
@@ -343,7 +343,7 @@ static int __devinit tegra_rtc_probe(str
 
 	/* set context info. */
 	info->pdev = pdev;
-	info->tegra_rtc_lock = __SPIN_LOCK_UNLOCKED(info->tegra_rtc_lock);
+	spin_lock_init(&info->tegra_rtc_lock);
 
 	platform_set_drvdata(pdev, info);
 
