---
 mm/highmem.c |    8 ++++++++
 1 file changed, 8 insertions(+)

Index: linux-rt.q/mm/highmem.c
===================================================================
--- linux-rt.q.orig/mm/highmem.c
+++ linux-rt.q/mm/highmem.c
@@ -209,6 +209,14 @@ static unsigned long pkmap_insert(struct
 	return vaddr;
 }
 
+/*
+ * Flush all unused kmap mappings in order to remove stray mappings.
+ */
+void kmap_flush_unused(void)
+{
+	WARN_ON_ONCE(1);
+}
+
 fastcall void *kmap_high(struct page *page)
 {
 	unsigned long vaddr;
