---
 arch/i386/kernel/hpet.c |    1 +
 1 file changed, 1 insertion(+)

Index: linux-rt.q/arch/i386/kernel/hpet.c
===================================================================
--- linux-rt.q.orig/arch/i386/kernel/hpet.c
+++ linux-rt.q/arch/i386/kernel/hpet.c
@@ -9,6 +9,7 @@
 #include <linux/pm.h>
 
 #include <asm/fixmap.h>
+#include <asm/i8253.h>
 #include <asm/hpet.h>
 #include <asm/i8253.h>
 #include <asm/io.h>
