---
 kernel/cpu.c |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-rt.q/kernel/cpu.c
===================================================================
--- linux-rt.q.orig/kernel/cpu.c
+++ linux-rt.q/kernel/cpu.c
@@ -37,12 +37,14 @@ void lock_cpu_hotplug(void)
 	struct task_struct *tsk = current;
 
 	if (tsk == recursive) {
+#ifdef CONFIG_PREEMPT_RT
 		static int warnings = 10;
 		if (warnings) {
 			printk(KERN_ERR "Lukewarm IQ detected in hotplug locking\n");
 			WARN_ON(1);
 			warnings--;
 		}
+#endif
 		recursive_depth++;
 		return;
 	}
