 kernel/rcutorture.c |    4 ++++
 1 file changed, 4 insertions(+)

Index: linux-rt.q/kernel/rcutorture.c
===================================================================
--- linux-rt.q.orig/kernel/rcutorture.c
+++ linux-rt.q/kernel/rcutorture.c
@@ -327,7 +327,11 @@ static void rcu_bh_torture_read_unlock(i
 
 static int rcu_bh_torture_completed(void)
 {
+#ifdef CONFIG_PREEMPT_RCU
+	return rcu_batches_completed();
+#else
 	return rcu_batches_completed_bh();
+#endif
 }
 
 static void rcu_bh_torture_deferred_free(struct rcu_torture *p)
