---
 kernel/latency_trace.c |    3 ---
 1 file changed, 3 deletions(-)

Index: linux-rt.q/kernel/latency_trace.c
===================================================================
--- linux-rt.q.orig/kernel/latency_trace.c
+++ linux-rt.q/kernel/latency_trace.c
@@ -40,9 +40,6 @@
 
 int trace_use_raw_cycles = 0;
 
-#define __raw_spinlock_t raw_spinlock_t
-#define need_resched_delayed() 0
-
 #ifdef CONFIG_EVENT_TRACE
 /*
  * Convert raw cycles to usecs.
