---
 arch/x86_64/kernel/smpboot.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-rt.q/arch/x86_64/kernel/smpboot.c
===================================================================
--- linux-rt.q.orig/arch/x86_64/kernel/smpboot.c
+++ linux-rt.q/arch/x86_64/kernel/smpboot.c
@@ -331,8 +331,8 @@ void __cpuinit start_secondary(void)
 	barrier();
 
 	/*
-  	 * Check TSC sync first:
- 	 */
+	 * Check TSC sync first:
+	 */
 	check_tsc_sync_target();
 
 	Dprintk("cpu %d: setting up apic clock\n", smp_processor_id()); 	
