 arch/i386/kernel/apic.c |    1 +
 1 file changed, 1 insertion(+)

Index: linux-2.6.23-rt1/arch/i386/kernel/apic.c
===================================================================
--- linux-2.6.23-rt1.orig/arch/i386/kernel/apic.c	2007-10-11 15:58:10.000000000 -0400
+++ linux-2.6.23-rt1/arch/i386/kernel/apic.c	2007-10-11 16:00:00.000000000 -0400
@@ -1307,6 +1307,7 @@ void smp_error_interrupt(struct pt_regs 
 	*/
 	printk (KERN_DEBUG "APIC error on CPU%d: %02lx(%02lx)\n",
 		smp_processor_id(), v , v1);
+	dump_stack();
 	irq_exit();
 }
 
