Subject: clockevents: Remove unused inline function

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>

---
 include/linux/clockchips.h |    1 -
 1 file changed, 1 deletion(-)

Index: linux-2.6.23-rt1/include/linux/clockchips.h
===================================================================
--- linux-2.6.23-rt1.orig/include/linux/clockchips.h	2007-10-11 15:58:16.000000000 -0400
+++ linux-2.6.23-rt1/include/linux/clockchips.h	2007-10-11 15:59:48.000000000 -0400
@@ -130,7 +130,6 @@ extern void clockevents_notify(unsigned 
 
 #else
 
-static inline void clockevents_resume_events(void) { }
 #define clockevents_notify(reason, arg) do { } while (0)
 
 #endif
