---
 net/core/dev.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.23-rt1/net/core/dev.c
===================================================================
--- linux-2.6.23-rt1.orig/net/core/dev.c	2007-10-11 15:57:49.000000000 -0400
+++ linux-2.6.23-rt1/net/core/dev.c	2007-10-11 16:00:35.000000000 -0400
@@ -1745,8 +1745,8 @@ int netif_rx_ni(struct sk_buff *skb)
 {
 	int err;
 
-	preempt_disable();
 	err = netif_rx(skb);
+	preempt_disable();
 	if (local_softirq_pending())
 		do_softirq();
 	preempt_enable();
