---
 arch/powerpc/kernel/irq.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.23-rt1/arch/powerpc/kernel/irq.c
===================================================================
--- linux-2.6.23-rt1.orig/arch/powerpc/kernel/irq.c	2007-10-11 15:58:06.000000000 -0400
+++ linux-2.6.23-rt1/arch/powerpc/kernel/irq.c	2007-10-11 16:00:05.000000000 -0400
@@ -113,7 +113,7 @@ static inline void set_soft_enabled(unsi
 	: : "r" (enable), "i" (offsetof(struct paca_struct, soft_enabled)));
 }
 
-void local_irq_restore(unsigned long en)
+void notrace local_irq_restore(unsigned long en)
 {
 	/*
 	 * get_paca()->soft_enabled = en;
