---
 drivers/net/loopback.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.23-rt1/drivers/net/loopback.c
===================================================================
--- linux-2.6.23-rt1.orig/drivers/net/loopback.c	2007-10-11 16:00:11.000000000 -0400
+++ linux-2.6.23-rt1/drivers/net/loopback.c	2007-10-11 16:00:59.000000000 -0400
@@ -159,7 +159,7 @@ static int loopback_xmit(struct sk_buff 
 	lb_stats->packets++;
 	put_cpu();
 
-	netif_rx(skb);
+	netif_rx_ni(skb);
 
 	return 0;
 }
