---
 arch/i386/kernel/process.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

Index: linux-2.6.23-rt1/arch/i386/kernel/process.c
===================================================================
--- linux-2.6.23-rt1.orig/arch/i386/kernel/process.c	2007-10-11 16:00:49.000000000 -0400
+++ linux-2.6.23-rt1/arch/i386/kernel/process.c	2007-10-11 16:00:50.000000000 -0400
@@ -136,7 +136,9 @@ EXPORT_SYMBOL(default_idle);
  */
 static void poll_idle (void)
 {
-	cpu_relax();
+	do {
+		cpu_relax();
+	} while (!need_resched() && !need_resched_delayed());
 }
 
 #ifdef CONFIG_HOTPLUG_CPU
