---
 drivers/net/8139too.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6.23-rt1/drivers/net/8139too.c
===================================================================
--- linux-2.6.23-rt1.orig/drivers/net/8139too.c	2007-10-11 15:59:57.000000000 -0400
+++ linux-2.6.23-rt1/drivers/net/8139too.c	2007-10-11 16:00:59.000000000 -0400
@@ -2128,10 +2128,10 @@ static int rtl8139_poll(struct net_devic
 		 * Order is important since data can get interrupted
 		 * again when we think we are done.
 		 */
-		local_irq_save(flags);
+		local_irq_save_nort(flags);
 		RTL_W16_F(IntrMask, rtl8139_intr_mask);
 		__netif_rx_complete(dev);
-		local_irq_restore(flags);
+		local_irq_restore_nort(flags);
 	}
 	spin_unlock(&tp->rx_lock);
 
