 kernel/rcutorture.c |    4 ++++
 1 file changed, 4 insertions(+)

Index: linux-2.6.23-rt1/kernel/rcutorture.c
===================================================================
--- linux-2.6.23-rt1.orig/kernel/rcutorture.c	2007-10-11 16:00:16.000000000 -0400
+++ linux-2.6.23-rt1/kernel/rcutorture.c	2007-10-11 16:00:17.000000000 -0400
@@ -390,7 +390,11 @@ static void rcu_bh_torture_read_unlock(i
 
 static int rcu_bh_torture_completed(void)
 {
+#ifdef CONFIG_PREEMPT_RCU
+	return rcu_batches_completed();
+#else
 	return rcu_batches_completed_bh();
+#endif
 }
 
 static void rcu_bh_torture_deferred_free(struct rcu_torture *p)
