---
 arch/i386/mm/highmem.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.23-rt1/arch/i386/mm/highmem.c
===================================================================
--- linux-2.6.23-rt1.orig/arch/i386/mm/highmem.c	2007-10-11 15:58:11.000000000 -0400
+++ linux-2.6.23-rt1/arch/i386/mm/highmem.c	2007-10-11 15:59:57.000000000 -0400
@@ -39,7 +39,7 @@ void *kmap_atomic_prot(struct page *page
 
 	idx = type + KM_TYPE_NR*smp_processor_id();
 	vaddr = __fix_to_virt(FIX_KMAP_BEGIN + idx);
-	BUG_ON(!pte_none(*(kmap_pte-idx)));
+	WARN_ON_ONCE(!pte_none(*(kmap_pte-idx)));
 	set_pte(kmap_pte-idx, mk_pte(page, prot));
 	arch_flush_lazy_mmu_mode();
 
