---
 kernel/time/timekeeping.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: linux-2.6.23-rt1/kernel/time/timekeeping.c
===================================================================
--- linux-2.6.23-rt1.orig/kernel/time/timekeeping.c	2007-10-11 16:01:13.000000000 -0400
+++ linux-2.6.23-rt1/kernel/time/timekeeping.c	2007-10-11 16:01:14.000000000 -0400
@@ -488,7 +488,7 @@ static void clocksource_adjust(s64 offse
  */
 void update_wall_time(void)
 {
-	cycle_t cycle_now;
+	cycle_t cycle_now, offset;
 
 	/* Make sure we're fully resumed: */
 	if (unlikely(timekeeping_suspended))
@@ -499,6 +499,7 @@ void update_wall_time(void)
 #else
 	cycle_now = clock->cycle_last + clock->cycle_interval;
 #endif
+	offset = (cycle_now - clock->cycle_last) & clock->mask;
 	clocksource_accumulate(clock, cycle_now);
 
 	clock->xtime_nsec += (s64)xtime.tv_nsec << clock->shift;
