---
 kernel/latency_trace.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.23-rt1/kernel/latency_trace.c
===================================================================
--- linux-2.6.23-rt1.orig/kernel/latency_trace.c	2007-10-11 16:00:07.000000000 -0400
+++ linux-2.6.23-rt1/kernel/latency_trace.c	2007-10-11 16:00:07.000000000 -0400
@@ -191,7 +191,7 @@ static int report_latency(cycle_t delta)
 /*
  * Number of per-CPU trace entries:
  */
-#define MAX_TRACE (65536UL*16UL)
+#define MAX_TRACE (65536UL)
 
 #define CMDLINE_BYTES 16
 
